components/feature/toolbar/src/main/java/mozilla/components/feature/toolbar/WebExtensionToolbarAction.kt

```diff
@@ -38,16 +45,32 @@ open class WebExtensionToolbarAction(
 
         rootView.setBackgroundResource(backgroundResource)
         padding?.let { rootView.setPadding(it) }
+
+        rootView.addOnAttachStateChangeListener(object : View.OnAttachStateChangeListener {
```

Comment:
Nice ✨ 
Similar to the [pattern used in](https://github.com/mozilla-mobile/android-components/blob/6fef6179e1f4bfe59359b86efa7035116b32237a/components/browser/icons/src/main/java/mozilla/components/browser/icons/BrowserIcons.kt#L179), it looks like this is a common need, maybe in the future we can have an utility extension for loading an image  async for everybody to use :)